CRM/ManualDirectDebit/Hook/PageRun/CustomGroupInjector.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ *  Provides 'Direct Debit Information' integration into 'Recurring Contribution Detail' view
+ */
+class CRM_ManualDirectDebit_Hook_PageRun_CustomGroupInjector {
+
+  /**
+   * Id of Contact
+   *
+   * @var int
+   */
+  private $contactId;
+
+  /**
+   * Id of Direct Debit Mandate Custom Group
+   *
+   * @var int
+   */
+  private $directDebitMandateCustomGroupId;
+
+  /**
+   * Mandate id for appropriate contribution
+   *
+   * @var int
+   */
+  private $appropriateMandateId;
+
+  /**
+   * Contribution id for appropriate recurring contribution
+   *
+   * @var int
+   */
+  private $appropriateContributionId;
+
+  public function __construct() {
+    $this->contactId = CRM_Utils_Request::retrieve('cid', 'Integer', $page, FALSE);
+    $this->directDebitMandateCustomGroupId = $this->getGroupIDbyName("direct_debit_mandate");
+
+    $recurrentContributionId = CRM_Utils_Request::retrieve('id', 'Integer', $page, FALSE);
+
+    $this->appropriateMandateId = $this->getAppropriateMandateId($recurrentContributionId);
+    $this->appropriateContributionId = $this->getAppropriateContributionId($recurrentContributionId);
+  }
+
+  /**
+   * Injects 'Direct Debit Information' custom group inside 'Recurring Contribution Detail' view
+   *
+   */
+  public function injectDirectDebitInformation() {
+    CRM_Core_Resources::singleton()
+      ->addScriptFile('uk.co.compucorp.manualdirectdebit', 'js/directDebitInformation.js')
+      ->addSetting([
+        'urlData' => [
+          'gid' => $this->directDebitMandateCustomGroupId,
+          'cid' => $this->contactId,
+          'recId' => $this->appropriateContributionId,
+          'mandateId' => $this->appropriateMandateId,
+        ],
+      ]);
+  }
+
+  /**
+   * Gets id of custom group by name
+   *
+   * @param $customGroupName
+   *
+   * @return int
+   */
+  private function getGroupIDByName($customGroupName) {
+    return civicrm_api3('CustomGroup', 'getvalue', [
+      'return' => "id",
+      'name' => $customGroupName,
+    ]);
+  }
+
+  /**
+   * Gets id of custom field by name
+   *
+   * @param $customFieldName
+   *
+   * @return int
+   */
+  private function getCustomFieldIdByName($customFieldName) {
+    return civicrm_api3('CustomField', 'getvalue', [
+      'return' => "id",
+      'name' => $customFieldName,
+    ]);
+  }
+
+  /**
+   * Gets id of mandate for recurrent contribution
+   *
+   * @param $recurrentContributionId
+   *
+   * @return int
+   */
+  private function getAppropriateMandateId($recurrentContributionId) {
```

Comment:
you have a type all over this class, u use **recurrent** instead of **current** .